zksync_sdk/wallet.py

```diff
@@ -77,26 +63,46 @@ async def build_change_pub_key(
             if eth_auth_type == ChangePubKeyTypes.ecdsa:
                 fee = await self.zk_provider.get_transaction_fee(FeeTxType.change_pub_key_ecdsa,
                                                                  self.address(),
-                                                                 fee_token)
+                                                                 fee_token.id)
             elif eth_auth_type == ChangePubKeyTypes.onchain:
                 fee = await self.zk_provider.get_transaction_fee(FeeTxType.change_pub_key_onchain,
                                                                  self.address(),
-                                                                 fee_token)
+                                                                 fee_token.id)
             elif eth_auth_type == ChangePubKeyTypes.create2:
                 fee = await self.zk_provider.get_transaction_fee(FeeTxType.change_pub_key_create2,
                                                                  self.address(),
-                                                                 fee_token)
-
+                                                                 fee_token.id)
             fee = fee.total_fee
         else:
-            fee = token.from_decimal(fee)
+            fee = fee_token.from_decimal(fee)
+
+        
+        change_pub_key, eth_signature = self.build_change_pub_key(fee_token,
+                                                                        fee, eth_auth_data,
+                                                                        nonce,
+                                                                        account_id,
+                                                                        valid_from,
+                                                                        valid_until)
+
+        return await self.send_signed_transaction(change_pub_key, eth_signature)
+
+    def build_change_pub_key(
```

Comment:
Since we are making breaking changes here we should probably confirm this with our partners/users. Also this interface makes it more difficult to create transactions to store and send them later (not right away) and we should consider if this is really necessary.